java/src/main/java/com/powsybl/python/SensitivityAnalysisResultContext.java

```diff
@@ -56,20 +59,42 @@ public SensitivityAnalysisResultContext(int flowRowCount, int flowColCount, int
         return contingencyId.isEmpty() ? baseCaseReferences : referencesByContingencyId.get(contingencyId);
     }
 
-    public PyPowsyblApiHeader.MatrixPointer createBranchFlowsSensitivityMatrix(String contingencyId) {
-        return createDoubleMatrix(() -> getValues(contingencyId), 0, flowRowCount, flowColCount);
+    public PyPowsyblApiHeader.MatrixPointer createBranchFlowsSensitivityMatrix(String matrixId, String contingencyId) {
+        SensitivityAnalysisContext.MatrixInfo m = this.branchFlowFactorsMatrix.get(matrixId);
+        return createDoubleMatrix(() -> getValues(contingencyId), m.getOffsetData(), m.getRowCount(), m.getColumnCount());
+    }
+
+    public PyPowsyblApiHeader.MatrixPointer createPreContingencyBranchFlowsSensitivityMatrix(String matrixId) {
+        SensitivityAnalysisContext.MatrixInfo m = this.preContingencyFactorMatrix.get(matrixId);
+        return createDoubleMatrix(() -> getValues(""), m.getOffsetData(), m.getRowCount(), m.getColumnCount());
+    }
+
+    public PyPowsyblApiHeader.MatrixPointer createPostContingencyBranchFlowsSensitivityMatrix(String matrixId, String contingency) {
+        SensitivityAnalysisContext.MatrixInfo m = this.postContingencyFactorMatrix.get(matrixId);
+        return createDoubleMatrix(() -> getValues(contingency), m.getOffsetData(), m.getRowCount(), m.getColumnCount());
+    }
+
+    public PyPowsyblApiHeader.MatrixPointer createBranchFlowsSensitivityMatrix(final String contingencyId) {
+        return createBranchFlowsSensitivityMatrix("default", contingencyId);
```

Comment:
Maybe the concept of "default" should only be on python side?

Reply:
In fact this dates back to the first implementation last year. It should be cleaned.